### PR TITLE
Add GitHub Pages support

### PR DESCRIPTION
**Pull request by travier**
Test deployment: https://travier.github.io/fcct/
Fixes: #128

### CODE REVIEW

**Comment by bgilbert** [comment]
It looks like the favicon uses the old CoreOS Inc. colors, not the new Fedora CoreOS colors.

File: docs/_config.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+title: coreos/fcct
+description: Fedora CoreOS Config Transpiler documentation
+baseurl: "/fcct"
+url: "https://coreos.github.io"
+# Comment above and use below for local development
+# url: "http://localhost:4000"
+permalink: /:title/
+
+remote_theme: pmarsceill/just-the-docs
```
Comment (bgilbert) [inline docs/_config.yml:9]: Should we be using a local fork?

File: docs/_config.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  - jekyll-remote-theme
+
+color_scheme: coreos
+
+# Aux links for the upper right navigation
+aux_links:
+  "FCCT on GitHub":
+    - "https://github.com/coreos/fcct"
+
+footer_content: "Copyright &copy; <a href=\"https://github.com/coreos\">CoreOS</a>."
```
Comment (bgilbert) [inline docs/_config.yml:20]: Red Hat?

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+layout: default
+nav_order: 1
+---
+
+# Fedora CoreOS Config Transpiler
+
+The Fedora CoreOS Config Transpiler (FCCT) translates human readable Fedora CoreOS Configs (FCCs)
+into machine readable [Ignition](https://github.com/coreos/ignition) Configs. See the [getting
+started](getting-started) guide for how to use FCCT and the [configuration spec](configuration-v1_0)
```
Comment (bgilbert) [inline docs/index.md]: 1_1.  I guess we should update the README too.

File: docs/examples.md [hunk 4]
```diff
@@ -180,5 +180,5 @@ systemd:
         WantedBy=multi-user.target
 ```
 
-[spec]: configuration-v1_0.md
+[spec]: specs.md
```
Comment (bgilbert) [inline docs/examples.md:183]: `specs.md` is an empty document.  That's okay on the website, with index generation, but isn't great for anyone browsing docs in the GitHub UI.

**Comment by travier** [comment]
> It looks like the favicon uses the old CoreOS Inc. colors, not the new Fedora CoreOS colors.

Hum indeed, good catch! Any idea where I could find the new one?

**Comment by travier** [comment]
> Hum indeed, good catch! Any idea where I could find the new one?

Found in fedora-websites repo.

File: docs/_config.yml (hunk 1 above)
Comment (travier) [reply docs/_config.yml:9]: Luca mentioned that too. I will set this up as we will have more and more depending on that.

File: docs/examples.md (hunk 4 above)
Comment (travier) [reply docs/examples.md:183]: Indeed. But I don't have any idea to make it better. Manual indexing?

Comment (travier) [reply docs/examples.md:183]: I will extend the specs page to make it more interesting

**Comment by travier** [comment]
Fixed all issues with the latest push.

File: docs/specs.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+Do not use the **experimental** specification for anything beyond **development
+and testing** as it is subject to change **without warning or announcement**.
+The following **experimental** specification version is currently available in
+`fcct`:
+
+- [v1.2.0-experimental](configuration-v1_2-exp.md)
+
+## FCCT specifications and Ignition specifications
+
+Each version of the FCCT specification usualy correspond to a version of the
```
Comment (bgilbert) [inline docs/specs.md:37]:
```suggestion
Each version of the FCCT specification corresponds to a version of the
```

File: docs/examples.md [hunk 6]
```diff
@@ -6,9 +12,9 @@ Here you can find a bunch of simple examples for using `fcct`, with some explana
 
 This example modifies the existing `core` user and sets its ssh key.
 
-```yaml fedora-coreos-config
```
Comment (bgilbert) [inline docs/examples.md:9]: These are how the `test` script knows to test the configs.  If we remove these, we'll need to find another solution for the script.

**Comment by travier** [comment]
I've pushed an alternative marker for the doc tests.

File: test [hunk 7]
```diff
@@ -25,8 +25,9 @@ for doc in docs/*md
 do
 	echo "Checking $doc"
 	# split each doc into a bunch of tmpfiles then run fcct on them
-	sed -n '/^```yaml fedora-coreos-config/,/^```/ p' < ${doc} \
-		| csplit - '/```yaml fedora-coreos-config/' '{*}' -z --prefix tmpdocs/tmpconfig -q
+	sed -n '/^<!-- fedora-coreos-config -->/,/^```/ p' < ${doc} \
+		| sed -n '/^<!-- fedora-coreos-config -->$/d' \
+		| csplit - '/```yaml/' '{*}' -z --prefix tmpdocs/tmpconfig -q
```
Comment (bgilbert) [inline test:30]: This isn't detecting any FCCs.

**Comment by travier** [comment]
The latest update should have fixed the remaining issues.

**Comment by travier** [comment]
Updated as I found a page missing in the doc.

**Comment by travier** [comment]
This should be good to merge. I don't have admin access to this repo so I can not do the next steps:
- GitHub pages setup
- Website URL setup in settings

**Comment by bgilbert** [comment]
Done!

**Pull request merged**